### ISSUE TITLE
valid-jsdoc: Return to accept `undefined` type

### ISSUE THREAD

**Issue by gyandeeps** [eslint/eslint#3382]
Eslint: 1.1.0
Parser: Default

**Code:**
```js
/* eslint valid-jsdoc: 2 */

/**
* hi
* @param {string} abc parammmmm
* @returns {undefined}
*/
function test(abc){
	var x = abc + 1;
}
```

**Output:**
```sh
3:1 - Missing JSDoc return description.
```

**Proposal:** It should allow `undefined` type as an acceptable return which doesn't require description as thats whats returned in javascript by default. Right now the rule support no description if we have `void` as type for return.

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by michaelficarra** [comment]
The documentation says you can use `requireReturn: false` to be allowed to omit `@returns` when there is no return statement in the function. Between that and `@returns {void}`, I don't see any need for this.

**Comment by gyandeeps** [comment]
Use case that we need `return` in jsdoc. It just when there is nothing than we should be able to use `undefined` as we can do `void`.

**Comment by michaelficarra** [comment]
What's the difference? Why can't you just change it to `void`?

**Comment by nzakas** [comment]
@michaelficarra it's not our position to judge preferences like this. We want to enable as many logical preferences as possible.

@gyandeeps feel free to make the change.

**Comment by michaelficarra** [comment]
But where do we stop? If I want to write `@returns {nothing}`, are we going to allow that as well, or are you going to tell me to use one of the existing options?

**Comment by vegarturo** [comment]
Running into the some problem here. 

>  or are you going to tell me to use one of the existing options?

I would like to know

**Comment by gyandeeps** [comment]
`undefined` and `void` both will be acceptable. Thats whats have been discussed so far.

**Comment by nzakas** [comment]
@michaelficarra you're going a little overboard here. Both `void` and `undefined` are part of the language, suggesting that we'd add `nothing` is just argumentative. It's this type of comment that is unnecessarily confrontational. Our job is to be as flexible as we can for our users, and sometimes that means having more than one way of doing things. I see this as a perfectly reasonable request given how the language works. Adding `nothing` has nothing to do with the language, so it would not be accepted.
